source/hic_hal/stm32/stm32h743ii/sdk.c

```diff
@@ -20,103 +20,118 @@
-//    RCC_OscInitTypeDef RCC_OscInitStruct = {0};
-
-//    SystemCoreClockUpdate();
-//    HAL_Init();
-
-//    /* Select HSI as system clock source to allow modification of the PLL configuration */
-//    RCC_ClkInitStruct.ClockType = RCC_CLOCKTYPE_SYSCLK;
-//    RCC_ClkInitStruct.SYSCLKSource = RCC_SYSCLKSOURCE_HSI;
-//    if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_1) != HAL_OK) {
-//        /* Initialization Error */
-//        util_assert(0);
-//    }
-
-//    /* Enable HSE bypass Oscillator, select it as PLL source and finally activate the PLL */
-//    RCC_OscInitStruct.OscillatorType = RCC_OSCILLATORTYPE_HSE;
-//    RCC_OscInitStruct.HSEState = RCC_CR_HSEON;
-//    RCC_OscInitStruct.PLL.PLLSource = RCC_PLLSOURCE_HSE;
-//    RCC_OscInitStruct.PLL.PLLState = RCC_PLL_ON;
-//    RCC_OscInitStruct.HSEPredivValue = RCC_HSE_PREDIV_DIV1;
-//    RCC_OscInitStruct.PLL.PLLMUL = RCC_PLL_MUL9;
-//    if (HAL_RCC_OscConfig(&RCC_OscInitStruct) != HAL_OK) {
-//        /* Initialization Error */
-//        util_assert(0);
-//    }
-
-//    /* Select the PLL as system clock source and configure the HCLK, PCLK1 and PCLK2 clocks dividers */
-//    RCC_ClkInitStruct.ClockType = (RCC_CLOCKTYPE_SYSCLK | RCC_CLOCKTYPE_HCLK | RCC_CLOCKTYPE_PCLK1 | RCC_CLOCKTYPE_PCLK2);
-//    RCC_ClkInitStruct.SYSCLKSource = RCC_SYSCLKSOURCE_PLLCLK;
-//    RCC_ClkInitStruct.AHBCLKDivider = RCC_SYSCLK_DIV1;
-//    RCC_ClkInitStruct.APB1CLKDivider = RCC_HCLK_DIV2;
-//    RCC_ClkInitStruct.APB2CLKDivider = RCC_HCLK_DIV1;
-//    if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_2) != HAL_OK) {
-//        /* Initialization Error */
-//        util_assert(0);
-//    }
-
-
-    pin_out_init(CONNECTED_LED_PORT, CONNECTED_LED_PIN_Bit);
-    CONNECTED_LED_PORT->BSRR = CONNECTED_LED_PIN;
-
-	//elee: toggle the pin, see the LED do something.
-		HAL_Delay(100);
-		CONNECTED_LED_PORT->BSRR = (CONNECTED_LED_PIN << 16);
-
+    RCC_OscInitTypeDef RCC_OscInitStruct = {0};
+    RCC_ClkInitTypeDef RCC_ClkInitStruct = {0};
+    RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {0};
+
+    SystemCoreClockUpdate();
+    HAL_Init();
+
+    /** Supply configuration update enable
+    */
+    HAL_PWREx_ConfigSupply(PWR_LDO_SUPPLY);
+    /** Configure the main internal regulator output voltage
+    */
+    __HAL_PWR_VOLTAGESCALING_CONFIG(PWR_REGULATOR_VOLTAGE_SCALE2);
+
+    while(!__HAL_PWR_GET_FLAG(PWR_FLAG_VOSRDY)) {}
+    /** Initializes the RCC Oscillators according to the specified parameters
+    * in the RCC_OscInitTypeDef structure.
+    */
+    RCC_OscInitStruct.OscillatorType = RCC_OSCILLATORTYPE_HSI48|RCC_OSCILLATORTYPE_HSI;
+    RCC_OscInitStruct.HSIState = RCC_HSI_DIV1;
+    RCC_OscInitStruct.HSICalibrationValue = RCC_HSICALIBRATION_DEFAULT;
+    RCC_OscInitStruct.HSI48State = RCC_HSI48_ON;
+    RCC_OscInitStruct.PLL.PLLState = RCC_PLL_ON;
+    RCC_OscInitStruct.PLL.PLLSource = RCC_PLLSOURCE_HSI;
+    RCC_OscInitStruct.PLL.PLLM = 4;
```

Comment:
Any clue what all these are? Are there defines you can use for these?

Reply:
do you mean the PLLM, PLLN, etc...? They are the different dividers and multipliers in PLL. There are no defines for the values. If anyone wants to tweak these, they'd need to go to the clock map anyways, so I don't think we need to comment about these.